bandit/core/manager.py

```diff
@@ -269,7 +270,11 @@ def run_tests(self):
                     self._show_progress("%s.. " % count, flush=True)
             try:
                 if fname == "-":
-                    sys.stdin = os.fdopen(sys.stdin.fileno(), "rb", 0)
+                    open_fd = os.fdopen(sys.stdin.fileno(), "rb", 0)
+                    sys.stdin = io.BytesIO(open_fd.read())
```

Comment:
So I'm 99% certain this is going to be a problem with some integration somewhere but if we were already patching this like this, perhaps it's fine? Would it not be better to do something like:

```py
stdin = io.BytesIO(sys.stdin.read().encode())
```

It's less code, and then we can pass `stdin` to `self._parse_file` instead of `sys.stdin`

Reply:
I agree patching sys.stdin looks really ugly.  To avoid that, I think we would need to pass a variable around in quite a few places.